src/lib/iptables.ts

```diff
@@ -0,0 +1,30 @@
+import * as Promise from 'bluebird';
+import * as _childProcess from 'child_process';
+
+const childProcess: any = Promise.promisifyAll(_childProcess);
+
+function clearAndAppendIptablesRule(rule: string): Promise<void> {
+	return childProcess.execAsync(`iptables -D ${rule}`)
+		.catchReturn()
+		.then(() => childProcess.execAsync(`iptables -A ${rule}`));
+}
+
+function clearAndInsertIptablesRule(rule: string): Promise<void> {
+	return childProcess.execAsync(`iptables -D ${rule}`)
+		.catchReturn()
+		.then(() => childProcess.execAsync(`iptables -I ${rule}`));
+}
+
+export function rejectOnAllInterfacesExcept(
+	allowedInterfaces: string[],
+	port: number,
+): Promise<void> {
+	// We delete each rule and create it again to ensure ordering (all ACCEPTs before the REJECT/DROP).
+	// This is especially important after a supervisor update.
+	return Promise.each(allowedInterfaces,
+		(iface) => clearAndInsertIptablesRule(`INPUT -p tcp --dport ${port} -i ${iface} -j ACCEPT`),
+	)
+		.then(() => clearAndAppendIptablesRule(`INPUT -p tcp --dport ${port} -j REJECT`))
```

Comment:
This indentation really confuses me, is there any way to keep it indented on the same level it actually executes at?

Reply:
This is the way that I've indented all of my typescript, 
this just looks slightly more weird due to the indented argument above. If this isn't acceptable I think we need to define exactly what to do in these situations, or just use prettier.